src/Dialect/ONNX/Rewrite.td

```diff
@@ -263,6 +278,55 @@ def FuseAddConvPattern: Pat<
    (RankXMinusRankYIs<1> $res, $y)]
 >;
 
+//===----------------------------------------------------------------------===//
+// This is to fuse the composition: 'Mul o Conv' into 'Conv' if the other input
+// of Mul is a constant, by multipling constant to 'w' of 'Conv':
+//
+// We have:
+//   (Conv)      z = i * w + b
+//   (Mul)       y = z x c      (where c is a constant)
+//
+// which corresponds to the following computation:
+//   y = i * new_w + b
+// where
+//   new_w = w x c
+//
+// The shape of 'c' must be compatible with that of 'w'
+//===----------------------------------------------------------------------===//
+
+def NormalizeMulPattern: Pat<
+  (ONNXMulOp $x, $y),
+  (ONNXMulOp $y, $x),
+  [(IsFromONNXConstantOp $x), (IsNotFromONNXConstantOp $y)]
+>;
+
+def FuseMulConvNullBiasPattern: Pat<
+  (ONNXMulOp:$res
+    (ONNXConvOp
+       $x, $w, $b, $auto_pad, $dilation, $group, $kernel_shape, $pads, $strides),
+    (ONNXConstantOp:$y $_, $denseAttr, $_, $_, $_, $_, $_, $_)),
+  (ONNXConvOp
+     $x,
+     // new_w
+     (ONNXConstantOpFromDenseAttr (ScaleByAxis<1> $w, $denseAttr)),
```

Comment:
I guess that `ScallByAxis` is to do multiplication of two constants. If so, it looks redundant. We can call something like`ONNXMulOp $w, (ONNXSqueeze $y, $axes)` and let `--constprop-onnx` do it job by folding `ONNXMulOp` and `ONNXSqueeze` into ONNXConstantOp, where `$axes` is rank(y) in order to add 1 as the last dimension.

Reply:
Done